src/serialmonitor/serialMonitor.ts

```diff
@@ -38,256 +30,170 @@ export class SerialMonitor implements vscode.Disposable {
-        const timestampFormat = await vscode.window.showInputBox();
-        if (!timestampFormat) {
-            Logger.warn("No timestamp format inputted, keeping previous timestamp format.");
-            return;
+    public async openSerialMonitor(restore: boolean = false): Promise<void> {
+        if (!this.currentPort) {
+            const ans = await vscode.window.showInformationMessage("No serial port was selected, please select a serial port first", "Yes", "No");
+            if (ans === "Yes") {
+                if (await this.selectSerialPort() === undefined) {
+                    return;
+                }
+            }
+            if (!this.currentPort) {
+                return;
+            }
         }
-        if (timestampFormat.indexOf("%") < 0) {
-            Logger.warn("Invalid timestamp format, keeping previous timestamp format.", { value: timestampFormat });
+
+        // if we're restoring, we want to use the most recent baud rate selected, rather than popping UI.
+        const baudRate = restore ? this.lastSelectedBaudRate : await this.selectBaudRate();
+
+        if (!baudRate) {
             return;
         }
-        if (!this._serialPortCtrl) {
-            Logger.warn("Serial Monitor has not been started.");
-            return;
+
+        try {
+            this.activePort = await this.serialMonitorApi.startMonitoringPort({
+                port: this.currentPort,
+                baudRate,
+                lineEnding: LineEnding.None,
+                dataBits: 8,
+                stopBits: StopBits.One,
+                parity: Parity.None,
+            });
+            this.activePort.onClosed(() => {
+                this.updatePortStatus(false);
+                this.activePort = undefined;
+            });
+            this.updatePortStatus(true);
+        } catch (err) {
+            vscode.window.showErrorMessage(`Error opening serial port: ${err.toString()}`);
         }
-        await this._serialPortCtrl.changeTimestampFormat(timestampFormat);
-        this._currentTimestampFormat = timestampFormat;
-        this._timestampFormatStatusBar.tooltip = `Timestamp Format: "${timestampFormat}"`;
     }
 
-    public async closeSerialMonitor(port: string, showWarning: boolean = true): Promise<boolean> {
-        if (this._serialPortCtrl) {
-            if (port && port !== this._serialPortCtrl.currentPort) {
-                // Port is not opened
-                return false;
-            }
-            const result = await this._serialPortCtrl.stop();
-            this.updatePortStatus(false);
-            return result;
-        } else if (!port && showWarning) {
-            Logger.notifyUserWarning("closeSerialMonitorError", new Error(constants.messages.SERIAL_PORT_NOT_STARTED));
-            return false;
-        }
+    public async closeSerialMonitor(port?: string): Promise<boolean> {
+        const closed = await this.serialMonitorApi.stopMonitoringPort(port ?? this.currentPort);

```

Comment:
Is it guaranteed that `this.currentPort` has a value?

Reply:
I believe it should be, but to cover corner cases, I'll check and ensure.